CoffeeShop.playground/Sources/CoffeeShopTest.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+import XCTest
+
+extension CoffeeShop: Equatable {
+    public static func == (lhs: CoffeeShop, rhs: CoffeeShop) -> Bool {
+        return lhs.name == rhs.name && lhs.location == rhs.location
+    }
+}
+
+public class CoffeeShopTest: XCTestCase {
+
+    public override func setUp() {
+    }
+
+    public override func tearDown() {
+    }
+
+    public func test_coffeeShop() {
+        let coffeeShop = CoffeeShop(name: "Starbucks Seattle", location: Location(x: 47.5809, y: -122.3160))
+        XCTAssertEqual(coffeeShop.name, "Starbucks Seattle")
+        XCTAssertEqual(coffeeShop.location, Location(x: 47.5809, y: -122.3160))
+    }
+
+    public func test_coffeeShopFromString() {
+        let coffeeShop = CoffeeShop(from: "Starbucks Seattle,-122.3160,47.5809", separator: ",")
+        XCTAssertNotNil(coffeeShop)
+        XCTAssertEqual(coffeeShop!.name, "Starbucks Seattle")
+        XCTAssertEqual(coffeeShop!.location, Location(x: 47.5809, y: -122.3160))
```

Comment:
@eandreea You should avoid using force unwrappers. I can very easily make this code crash. Is better to have a guard and fail earlier. Ex
guard let sut = coffeeShop else {
 // fail
}
XCTAssertEqual(sut.name, "Starbucks Seattle")

Reply:
Yeah, big mistake on my part, this was due to my forgetting to put continueAfterFailure = false in the test